src/GameOptions/ui/ConnectionBauble.tsx

```diff
@@ -0,0 +1,41 @@
+import React from "react";
+
+interface baubleState {
+  connection: boolean;
+  callback: () => boolean;
+}
+
+interface baubleProps {
+  callback: () => boolean;
+}
+
+export class ConnectionBauble extends React.Component<baubleProps> {
```

Comment:
Just for consistency, I believe the rest of the project only uses functional React components rather than class components.

Reply:
The component has a timer within. I wasn't able to find a way to do anything with a timer (e.g. internal state) when using functional components. If someone is able to point me to how something like that is possible or an alternative method to have a connectionBauble without internal state, I'd be happy to change it.

Reply:
Update: I figured it out pretty quickly after having another look. New commit is with a functional component instead!

Reply:
Nice! Yeah you can see why functional components are preferred, way less code for the same thing.

A few issues though. You're never clearing the interval you set in useEffect. useEffect should return a callback function which is ran when the component unmounts. Also you're still referencing this.state.connection, when it is just connection now.

There's also an error on gameOptionsPage: `import { ConnectionBaub, ConnectionBauble } from "./ConnectionBauble";`

I believe by convention interfaces/types usually are capitalized so they aren't confused for regular variables.

This is what I got while converting it to a functional component (actually I had left in a tick function, but I just removed it after I saw the way got rid of it/it wasn't needed):

```typescript
import React, { useEffect, useState, ReactElement } from "react";

interface BaubleProps {
  callback: () => boolean;
}

export function ConnectionBauble(props: BaubleProps): ReactElement {
  const [connection, setConnection] = useState(props.callback());
  useEffect(() => {
    const timerID = setInterval(() => setConnection(props.callback()), 1000);
    return () => clearInterval(timerID);
  });

  return <span>connection ? "Connected" : "Disconnected"</span>;
}
```